Cython/Includes/cpython/datetime.pxd

```diff
@@ -84,14 +168,27 @@ cdef extern from "datetime.h":
         PyTypeObject *TZInfoType
 
         # constructors
-        object (*Date_FromDate)(int, int, int, PyTypeObject*)
-        object (*DateTime_FromDateAndTime)(int, int, int, int, int, int, int, object, PyTypeObject*)
-        object (*Time_FromTime)(int, int, int, int, object, PyTypeObject*)
-        object (*Delta_FromDelta)(int, int, int, int, PyTypeObject*)
+        date (*Date_FromDate)(int, int, int, PyTypeObject*)
+        datetime (*DateTime_FromDateAndTime)(int, int, int, int, int, int, int, object, PyTypeObject*)
+        time (*Time_FromTime)(int, int, int, int, object, PyTypeObject*)
+        timedelta (*Delta_FromDelta)(int, int, int, int, PyTypeObject*)
 
         # constructors for the DB API
-        object (*DateTime_FromTimestamp)(object, object, object)
-        object (*Date_FromTimestamp)(object, object)
+        datetime (*DateTime_FromTimestamp)(object, object, PyObject*)
+        date (*Date_FromTimestamp)(object, object)
```

Comment:
Let's make these two match the other constructors, and pass the type as type pointer.
```suggestion
        datetime (*DateTime_FromTimestamp)(PyTypeObject*, object, PyObject*)
        date (*Date_FromTimestamp)(PyTypeObject*, object)
```